Update migration docs (#305)
TheThingsIndustries/lorawan-stack-docs: doc/content/getting-started/migrating/major-changes.md

---

Before going in detail with the migration process, this section will give you a breakdown of differences between {{% ttnv2 %}} and {{% tts %}}, along with some guidelines to make the migration process easier to manage.

## Architecture

{{% tts %}} is using a new architecture. See [Components]({{< ref "/reference/components" >}}) for a complete overview of the different components of {{% tts %}}.

## LoRaWAN support

{{% tts %}} requires the LoRaWAN version (MAC) and Regional Parameters version (LoRaWAN PHY version) to be set per end device. These default to LoRaWAN version **MAC V1.0.2** and LoRaWAN Regional Parameters version **PHY V1.0.2 Rev B** for end devices imported from {{% ttnv2 %}}, because this configuration is the most commonly used one.

End devices also need to be set with a frequency plan. You will have to choose the frequency plan corresponding to your region. See [Frequency Plans]({{< ref "/reference/frequency-plans" >}}) for a list of supported Frequency Plans and their respective IDs.

Furthermore, {{% tts %}} brings full support for all LoRaWAN versions, as well as modes to support class B and class C capabilities. 

> ## Rx1Delay

Reviewer: ```suggestion
## RX1 Delay
```

> Devices imported from {{% ttnv2 %}} are configured with an `Rx1Delay` of 1 second, by default. In {{% tts %}} we recommend using an `Rx1Delay` of 5 seconds to accommodate for high latency backhauls and/or [peering with Packet Broker]({{< ref "/reference/packet-broker" >}}). 
> 
> See the [MAC settings guide]({{< ref "/devices/mac-settings" >}}) for more information and instructions about configuring `Rx1Delay`.

Reviewer: ```suggestion
The RX1 delay is the time after which the first receive window (RX1) opens. In the receive window, the Network Server can schedule downlink. The second receive window (RX2) opens exactly one second after the first opens. In LoRaWAN, the RX1 delay can be between 1 and 15 seconds.

Devices imported from {{% ttnv2 %}} are configured with an RX1 delay of 1 second. In all {{% tts %}} deployments, the recommended RX1 delay is 5 seconds to accommodate for high latency backhauls and/or [peering with Packet Broker]({{< ref "/reference/packet-broker" >}}). 

See the [MAC settings guide]({{< ref "/devices/mac-settings" >}}) for more information and instructions about configuring the RX1 delay.
```

## MAC Commands

{{% tts %}} expects that all end devices comply with the LoRaWAN specification by default, which means that the end devices should respond to Network Server MAC command requests accordingly. If a device fails to answer a MAC Command in a timely manner, there may be disruptions to the device uplink or downlink traffic. As mentioned in the LoRaWAN specification, the Network Server of {{% tts %}} will always prioritize MAC commands over application payloads on downlink.

In case a device is not fully compliant with the LORaWAN specification, it can still work on {{% tts %}}, but it may require custom MAC settings configuration.

### DevStatusReq

The `DevStatusReq` MAC command is sent by the Network Server periodically to check the current status of the end device. Devices are expected to send a `DevStatusAns` reply.

> For end devices that ignore this MAC command, make sure to configure the `StatusTimePeriodicity` (time duration after which a `DevStatusReq` is issued by the Network Server) and `StatusCountPeriodicity` (number of frames after which a `DevStatusReq` is issued) of the device explicitly to `0`. 

Reviewer: What are these terms `StatusTimePeriodicity` and `StatusCountPeriodicity`? They are not TTS terms are they? Are they literally LoRaWAN?

Author: I have no idea, this is the existing content that someone else wrote. I was just tweaking it a bit

Migrating the {{% ttnv2 %}} payload encoder and decoder to an uplink and downlink payload formatter should be straightforward. 

See [Payload Formatters]({{< ref "/integrations/payload-formatters" >}}) for more details.

### Integrations

Read the full documentation dedicated to [Integrations]({{< ref "/integrations" >}}) supported by {{% tts %}}.

#### MQTT Traffic

{{% tts %}} has a new MQTT server address, so you will need to change this address in your application accordingly. You will also need to create API keys and update your MQTT credentials. 

Read more in the [MQTT Server]({{< ref "/integrations/mqtt" >}}) section.

#### Webhooks

The HTTP Integration is now called HTTP Webhooks. The payload format, downlink endpoints, paths and security settings have changed.

Read more in the [HTTP Webhooks]({{< ref "/integrations/webhooks" >}}) section.

#### Storage Integration

> {{% tts %}} supports a Storage integration similar to {{% ttnv2 %}}. 

Reviewer: ```suggestion
{{% tts %}} supports a Storage Integration similar to {{% ttnv2 %}}. 
```
